### PR TITLE
Add metric for tracking our success rate with oauth

### PR DESCRIPTION
**Pull request by Half-Shot**
(Also fix a small bug around the blocked metric name)

### CODE REVIEW

File: src/OAuth2.ts [hunk 1]
```diff
@@ -113,6 +117,10 @@ export class OAuth2 {
         // expects inbound roomIds to be 32 chars.
         const token = uuid().substr(0, INTERNAL_ID_LEN);
         this.userTokensWaiting.set(token, userId);
+        setTimeout(() => {
+            this.main.incCounter(METRIC_OAUTH_SESSIONS, {result: "failed", reason: "timeout"});
```
Comment (Half-Shot) [inline src/OAuth2.ts:121]: I should log something here...

File: src/Main.ts [hunk 2]
```diff
@@ -408,16 +410,22 @@ export class Main {
             name: METRIC_PUPPETS,
         });
         const bridgeBlocked = prometheus.addGauge({
-            name: "bridge_blocked",
+            name: "blocked",
```
Comment (tadzik) [inline src/Main.ts:413]: Is that the bug? What's wrong with `bridge_blocked`?

Comment (Half-Shot) [reply src/Main.ts:413]: `prometheus.addGauge` prefix's with bridge_ by default, so you end up with bridge_bridge_blocked :p

my bad for not catching the first time.

Comment (tadzik) [reply src/Main.ts:413]: Oh on /o\ Good catch!

**Review by tadzik — APPROVED** [review]

**Pull request merged**